compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp

```diff
@@ -105,6 +105,14 @@ static bool areFusableOps(MLIRContext *context, OpOperand *fusedOperand,
   if (!producerOp->hasOneUse())
     return false;
 
+  // Do no fuse dequantization-like operations with producers as we want to keep
+  // the smallest bitwidths at the dispatch boundaries, unless the consumer
+  // dequantization op only has one use, in which case elementwise op fusion
+  // is fine.
+  if (isDequantizationLikeOp(consumerOp) && !consumerOp->hasOneUse()) {
```

Comment:
The first test below tries to illustrate my thought process
```
// CHECK-LABEL: util.func public @fuse_single_use_dequant_with_producer
//       CHECK:   %[[GENERIC0:.+]] = linalg.generic
//       CHECK:     arith.subf
//  CHECK-NEXT:     arith.extf
//  CHECK-NEXT:     math.exp
```
Where I was thinking that in simple single use chains, fusing with the producer `arith.subf` just adds an extra input to either the producer or consumer that this elementwise op gets fused with. Thinking about it again, you're probably right though, it seems as if we're using "dequantization-like" as a proxy for making rematerialization decisions so this check shouldn't be necessary.

In that case, we probably need to review fusion through reshapes again and make it work a bit better with Falcon7b to avoid regressing the dispatch count there so much.